cypress/utils/Util.ts

```diff
@@ -0,0 +1,114 @@
+export function uniqueName(prefix: string) {
+  if (isEnvironmentType(EnvType.Mocked)) {
+    return prefix;
+  } else if (isEnvironmentType(EnvType.Dev)) {
+    return `testui-${prefix}-${Cypress._.uniqueId(Date.now().toString())}`;
+  } else {
+    throw new Error(
+      "The environment type is not recognized. Values of CYPRESS_TEST_TYPE are defined by the 'EnvType' enum."
+    );
+  }
+}
+
+export function pageWasLoaded() {
+  cy.get("#nav-toggle", { timeout: 30000 }).should("be.visible");
+  cy.ouiaId("loading-table", "PF4/Card").should("not.exist");
+}
+
+export function createInstance(newInstanceName: string, action?: string) {
+  cy.ouiaId("create-smart-event-instance", "PF4/Button").click();
+  cy.ouiaId("create-instance", "PF4/ModalContent").then(($modal) => {
+    cy.wrap($modal)
+      .should("be.visible")
+      .ouiaId("cloud-provider", "skeleton")
+      .should("not.exist");
+    cy.wrap($modal).within(() => {
+      cy.ouiaId("cloud-region", "PF4/Select", { timeout: 20000 }).should(
+        "not.have.text",
+        "Select Region"
+      );
+      cy.ouiaId("new-name", "PF4/TextInput").type(newInstanceName);
+      cy.ouiaId("info-instance-available-soon", "PF4/Alert").should(
+        "have.text",
+        "Info alert:Your Smart Events instance will be ready for use shortly after creation."
+      );
+      cy.ouiaId(action ?? "submit", "PF4/Button").click();
+    });
+    cy.wrap($modal, { timeout: 20000 }).should("not.exist");
+  });
```

Comment:
I found out that we can use aliases instead of `then`. Here is the example of such code:
```
  cy.ouiaId("create-smart-event-instance", "PF4/Button").click();
  cy.ouiaId("create-instance", "PF4/ModalContent").as("modal");
  cy.get("@modal")
    .should("be.visible")
    .ouiaId("cloud-provider", "skeleton")
    .should("not.exist");
  cy.get("@modal")
    .ouiaId("cloud-region","PF4/Select", {timeout: 20000}).should("not.have.text", "Select Region");
  cy.get("@modal").within(() => {
    //TODO: inspect why the DOM is detached here
    cy.ouiaId("new-name", "PF4/TextInput").as("new-name");
    cy.get("@new-name").type(newInstanceName);
    cy.ouiaId("info-instance-available-soon", "PF4/Alert").as("info");
    cy.get("@info").should(
      "have.text",
      "Info alert:Your Smart Events instance will be ready for use shortly after creation."
    );
    cy.ouiaId(action ?? "submit", "PF4/Button").click();
  });
  cy.get("@modal", { timeout: 20000 }).should("not.exist");
```

Reply:
@sjamboro I have no strong opinion. I am not familiar with life-cycle of aliases. It could bring more problems than benefits. I [found out](https://www.youtube.com/watch?v=VPznmFpa1Jc) that it could be useful to avoid the `detached from DOM` error. So I rewrite it and find out that code is better readable for me. But aliases were not useful (test fails again but not so offten) then I have defined `pageWasLoaded` and it fixed all occurence of the `detached from DOM` error. So let leave decision on a creator of a test.

Reply:
@jomarko I have try to put some description here: https://github.com/sjamboro/sandbox-ui/blob/MGDOBR-865/cypress/README.md#how-to-create-tests

Should I rewrite it or highligths important information?

Reply:
@sjamboro I think it is fine to have a description/guide on place you pointed to, resolving this conversation, thank you

Reply:
I found code is readable in both the cases (then and as ). To me , it is hardly making any difference from reading point of view.
But after reading links shared by Jozef, i agree with him that as seems not be much beneficial